xcell/cls/cov.py

```diff
@@ -105,6 +111,10 @@ def get_covariance(self):
                                       cla1b1, cla1b2, cla2b1, cla2b2,
                                       wa, wb)
 
+        if self.nl_marg:
+            _, nl = self.clA1A2.get_ell_nl
+            cov += self.nl_prior**2 * (nl[:, None] * nl[None, :])
```

Comment:
Will this work? This will have 3 axis instead of two. `nl.shape = (dof, number_of_bpw)`